### PR TITLE
Memory Footprint Reduction

### PR DESCRIPTION
**Pull request by hguermaz**
## Description
- Removed the ME distortions from the Parent_PCS.
- Moved restoration buffer(s) from the Parent_PCS to the Child_PCS.
- Moved rate estimation buffer(s) from the Child_PCS to the EncDec_Context if 1 EncDec_Segment.
- Moved EncDec only (not EC) MD results from the Child_PCS to the EncDec_Context.

## Author(s)
@chkngit 
@hguermaz 

## Testing and Performance
- ~28% Memory Footprint reduction for 1080p Lp-1 / Look-ahead OFF 
- Identical streams in M0/M1.

### CODE REVIEW

**Comment by hassount** [comment]
@hguermaz 
can you please rebase this PR

**Comment by hguermaz** [comment]
@hassount rebased.

File: Source/API/EbSvtAv1Enc.h [hunk 1]
```diff
@@ -377,6 +377,11 @@ typedef struct EbSvtAv1EncConfiguration {
     * Default is -1. */
     int enable_intra_edge_filter;
 
+    /* Picture based rate estimation
+    *
+    * Default is - 1. */
+    int pic_based_rate_est;
```
Comment (mafonsoN) [inline Source/API/EbSvtAv1Enc.h]: Can you give more details about this option?

**Comment by hguermaz** [comment]
@mafonsoN This signal is used to enable picture based rate estimation, and we are taking advantage of this input to reduce the memory footprint further for the lp 1 test case (where only 1 segment is used @ EncDec Process).

**Pull request merged**